File: backend/python/app/models/todo.py
from datetime import date, datetime
from sqlalchemy import Boolean, Date, DateTime, Integer, String, func
from sqlalchemy.orm import Mapped, mapped_column
from app.db.base import Base


class Todo(Base):
    __tablename__ = "Todo"

    id: Mapped[int] = mapped_column(Integer, primary_key=True, autoincrement=True)
    text: Mapped[str] = mapped_column("text", String(500))
    description: Mapped[str | None] = mapped_column("description", String(2000), nullable=True)
    completed: Mapped[bool] = mapped_column("completed", Boolean, default=False)
    priority: Mapped[str] = mapped_column("priority", String, default="medium")
    due_date: Mapped[date | None] = mapped_column("dueDate", Date, nullable=True)
    created_at: Mapped[datetime] = mapped_column(
        "createdAt", DateTime(timezone=True), server_default=func.now()
    )
    updated_at: Mapped[datetime] = mapped_column(
        "updatedAt", DateTime(timezone=True), server_default=func.now(), onupdate=func.now()
    )

File: backend/donet/TodoApi/Models/Entities/Todo.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace TodoApi.Models.Entities;

public class Todo
{
    [Key]
    [Column("id")]
    public int Id { get; set; }

    [Required]
    [MaxLength(500)]
    [Column("text")]
    public string Text { get; set; } = string.Empty;

    [MaxLength(2000)]
    [Column("description")]
    public string? Description { get; set; }

    [Column("completed")]
    public bool Completed { get; set; }

    [MaxLength(20)]
    [Column("priority")]
    public string Priority { get; set; } = "medium";

    [Column("dueDate")]
    public DateTime? DueDate { get; set; }

    [Column("createdAt")]
    public DateTime CreatedAt { get; set; }

    [Column("updatedAt")]
    public DateTime UpdatedAt { get; set; }
}